x-pack/plugins/snapshot_restore/public/application/components/restore_snapshot_form/steps/step_logistics/step_logistics.tsx

```diff
@@ -343,15 +425,15 @@ export const RestoreSnapshotStepLogistics: React.FunctionComponent<StepProps> =
             <h3>
               <FormattedMessage
                 id="xpack.snapshotRestore.restoreForm.stepLogistics.renameIndicesTitle"
-                defaultMessage="Rename indices"
+                defaultMessage="Rename indices and data streams"
```

Comment:
Alphabetical order
```suggestion
                defaultMessage="Rename data streams and indices"
```

I'm not sure how to present this visually, but I think that it may be worth notifying users that renaming a data stream also renames it's backing indices. For example, if you rename the `logs` data stream to `restored-logs`, the backing index `.ds-logs-000005` is renamed to `.ds-restored-logs-000005`.

It may also be good to reiterate that the new data stream name must have a matching index template. Renaming a data stream could break this matching. If no matching index template exists, the data stream cannot rollover or create new backing indices.

Reply:
For this particular instance, you're already using a EuiDescribedFormRow, which gives you space to add descriptive text. There’s no need to put text in a popover, just put it directly in the description of the form row which right now is just a single sentence.